src/litdata/streaming/dataloader.py

```diff
@@ -631,13 +635,19 @@ def __init__(
 
     def __iter__(self) -> Any:
         if not self.restore:
-            self._latest_worker_idx = 0
-            self._worker_idx = cycle(list(range(self.num_workers if self.num_workers > 0 else 1)))
-            self._worker_idx_iter = iter(self._worker_idx)
+            if (
+                not isinstance(self.dataset, ParallelStreamingDataset)
+                or self.dataset._length is None
+                or self.current_epoch == 0
+            ):
+                self._latest_worker_idx = 0
```

Comment:
Better to wrap those blocks of code into method, so they can be tested and extended with more ease.